### PR TITLE
Process String Array parameters in Endpoints

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Internal JS-5234

*Description of changes:*
Process String Array parameters in Endpoints

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by trivikr** [comment]
There's no diff in codegen, since no models contain StringArray parameters.

Testing will be done in [util-endpoints/src/__mocks__](https://github.com/smithy-lang/smithy-typescript/tree/main/packages/util-endpoints/src/__mocks__), similar to https://github.com/smithy-lang/smithy-typescript/pull/1374

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java [hunk 1]
```diff
@@ -102,6 +107,12 @@ public String defaultAsCodeString() {
             case "boolean":
                 buffer += paramNode.expectBooleanMember("default").getValue() ? "true" : "false";
                 break;
+            case "StringArray":
```
Comment (sugmanue) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java:110]: Minor, this case can be removed. This name in particular is not case sensitive, see [here](https://github.com/smithy-lang/smithy/blob/5da00e8e500175e75d35703e623961671c785d9d/smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/language/syntax/parameters/ParameterType.java#L54)

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java [hunk 2]
```diff
@@ -64,6 +65,10 @@ public ParameterGenerator(String key, Node param, boolean isInputKey) {
                 case "boolean":
                     tsParamType = "boolean";
                     break;
+                case "StringArray":
```
Comment (sugmanue) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java:68]: Minor, this case can be removed. This name in particular is not case sensitive, see [here](https://github.com/smithy-lang/smithy/blob/5da00e8e500175e75d35703e623961671c785d9d/smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/language/syntax/parameters/ParameterType.java#L54)

**Review by sugmanue — APPROVED** [review]

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java (hunk 2 above)
Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java]: Updated in [`c62112d` (#1376)](https://github.com/smithy-lang/smithy-typescript/pull/1376/commits/c62112d23e770d152af6b98f77e7f53c62865910)

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java (hunk 1 above)
Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java]: Updated in https://github.com/smithy-lang/smithy-typescript/pull/1376/commits/c62112d23e770d152af6b98f77e7f53c62865910

**Review by sugmanue — APPROVED** [review]

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/RuleSetParameterFinder.java [hunk 3]
```diff
@@ -90,6 +106,11 @@ public Map<String, String> getStaticContextParamValues(OperationShape operation)
                     value = "`" + definition.getValue().expectStringNode().toString() + "`";
                 } else if (definition.getValue().isBooleanNode()) {
                     value = definition.getValue().expectBooleanNode().toString();
+                } else if (definition.getValue().isArrayNode()) {
+                    ArrayNode arrayNode = definition.getValue().expectArrayNode();
+                    value = arrayNode.getElements().stream()
+                      .map(element -> "'" + element.expectStringNode().getValue() + "'")
```
Comment (kuhe) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/RuleSetParameterFinder.java:112]: Probably need string escape detection here. Also recommend using triple quotes in Java to enable double quotes in JS.

**Review by kuhe — APPROVED** [review]

**Pull request merged**